db/db_impl/db_impl.cc

```diff
@@ -5631,6 +5630,91 @@ Status DBImpl::CreateColumnFamilyWithImport(
   return status;
 }
 
+Status DBImpl::ClipDB(ColumnFamilyHandle* column_family, const Slice& begin_key,
+                      const Slice& end_key) {
+  assert(column_family);
+  Status status;
+
+  {
+    // Lock db mutex
+    InstrumentedMutexLock l(&mutex_);
+    // Flush memtable
+    FlushOptions flush_opts;
+    flush_opts.allow_write_stall = true;
+    auto* cfd =
+        static_cast_with_check<ColumnFamilyHandleImpl>(column_family)->cfd();
+    if (immutable_db_options_.atomic_flush) {
+      mutex_.Unlock();
+      status = AtomicFlushMemTables(flush_opts, FlushReason::kGetLiveFiles,
```

Comment:
We should use kOthers or kDeleteFiles for flush reason. I think `entered_write_thread` should be false since we are not in write thread.